# Controller Tunning Guide
# 1. Get the correct hover throttle (most important)
# 2. Start with P control for both position and velocity. Start with smaller value (increase no more than 20%)
# 3. If z direction position has steady-state error, give a little p_i.
# 4. I don't think it is necessary to use position d and velocity i and d.
#

# ROS Topic
# odom_topic: /Odometry
# for real world, use LIO
odom_topic: /Odom_high_freq
# for sim, use the vision pose
# odom_topic: /iris_0/mavros/vision_odom/odom
# imu_topic: /lidar/imu  // lidar imu 
# for real world, use mavros
imu_topic: /mavros/imu/data  
# imu_topic: /iris_0/mavros/imu/data
target_topic: /planning/pos_cmd
# target_topic: /planning/pos_cmd

# choose the right namespce
# attitude_target_topic: /iris_0/mavros/setpoint_raw/attitude
attitude_target_topic: /mavros/setpoint_raw/attitude
# position_arget_topic: /iris_0/mavros/setpoint_raw/local
position_arget_topic: /mavros/setpoint_raw/local

# Log
# ~~~~~~~~~~~~~~~~~NOTE~~~~~~~~~~~~~~~~~~~~~~
# change to your ws
save_path: ~/experiment/pid_ctrl/pid_tracking_data.csv
# ~~~~~~~~~~~~~~~~~NOTE~~~~~~~~~~~~~~~~~~~~~~


# body
use_fcu_ctrl: false      # use FCU control
body_rate_control: true # all parameters
attitude_control: false # PID and hover thrust
acceleration_control: false # only PID matters


# Position（外环）
position_p: [1.0, 1.00, 4.0]  # X/Y 提高响应
# Velocity（内环）
velocity_p: [1.8, 1.8, 4.0]  # 提高X/Y响应，Z略提

# Yaw 
yaw_p: 0.8
yaw_i: 0.1
yaw_d: 0.1

# Body rate control
attitude_control_roll_pitch: 3  # (this will not be used if attitude control is true)
attitude_control_yaw: 3         # (this will not be used if attitude control is true)
hover_thrust: 0.25             # this need to be very accurate for a good control (most important)

# Error limit
pxy_error_max: 0.6
vxy_error_max: 1.0
pz_error_max: 0.3
vz_error_max: 0.75

# display message
verbose: true

# Ignore
IGNORE_ACC_VEL: false
IGNORE_ACC: false


# For Circle 
circle_radius: 3.0 
time_to_max_radius: 30
yaw_control: True
velocity: 2.5
takeoff_height: 1.0 
max_path_length: 600

# log

